clang/lib/CIR/Dialect/Transforms/FlattenCFG.cpp

```diff
@@ -906,6 +911,72 @@ class CIRTernaryOpFlattening : public mlir::OpRewritePattern<cir::TernaryOp> {
   }
 };
 
+struct UnifyReturn : public OpRewritePattern<ReturnOp> {
+  using OpRewritePattern<ReturnOp>::OpRewritePattern;
+
+  UnifyReturn(MLIRContext *context, cir::FuncOp func, Block *retBlock)
+      : OpRewritePattern<ReturnOp>(context), func(func), retBlock(retBlock) {}
+
+  mlir::LogicalResult
+  matchAndRewrite(cir::ReturnOp ret,
+                  mlir::PatternRewriter &rewriter) const override {
+    mlir::OpBuilder::InsertionGuard guard(rewriter);
+    auto fn = ret->getParentOfType<cir::FuncOp>();
+    if (!fn || fn != func)
+      return mlir::failure();
+    // Replace 'return' with 'br <retBlock>'
```

Comment:
It looks like according to the coding style all the comments should be period-terminated sentences.